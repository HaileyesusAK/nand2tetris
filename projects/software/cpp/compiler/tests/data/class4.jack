class Circle {
	field int radius;
	method int diameter() {
		return 2 * radius;	
	}
}
