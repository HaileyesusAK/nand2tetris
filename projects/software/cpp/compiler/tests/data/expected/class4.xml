<class>
    <keyword> class </keyword>
    <identifier> Circle </identifier>
    <symbol> { </symbol>
    <classVarDec>
        <keyword> field </keyword>
        <keyword> int </keyword>
        <identifier> radius </identifier>
        <symbol> ; </symbol>
    </classVarDec>
    <subroutineDec>
        <keyword> method </keyword>
        <keyword> int </keyword>
        <identifier> diameter </identifier>
        <symbol> ( </symbol>
        <parameterList>
        </parameterList>
        <symbol> ) </symbol>
        <subroutineBody>
            <symbol> { </symbol>
            <statements>
                <returnStatement>
                    <keyword> return </keyword>
                    <expression>
                        <term>
                            <integerConstant> 2 </integerConstant>
                        </term>
                        <symbol> * </symbol>
                        <term>
                            <identifier> radius </identifier>
                        </term>
                    </expression>
                    <symbol> ; </symbol>
                </returnStatement>
            </statements>
            <symbol> } </symbol>
        </subroutineBody>
    </subroutineDec>
    <symbol> } </symbol>
</class>
